express-crud/server.ts

```diff
@@ -0,0 +1,139 @@
+import express from 'express';
+import pg from 'pg';
+import { ClientError, errorMiddleware } from './lib/index.js';
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    // Allow non-SSL traffic to localhost
+    rejectUnauthorized: false,
+  },
+});
+
+const app = express();
+
+app.use(express.json());
+
+app.get('/api/grades', async (req, res, next) => {
+  try {
+    const sql = `
+    select *
+    from "grades";
+    `;
+    const result = await db.query(sql);
+    res.status(200).json(result.rows);
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.get('/api/grades:gradeId', async (req, res, next) => {
+  try {
+    const { gradeId } = req.params;
+    if (gradeId === undefined) {
+      throw new ClientError(400, 'gradeId is required');
+    }
+    const sql = `
+    select *
+    from "grades"
+    where "gradeId" = $1
+    `;
+    const result = await db.query(sql, [gradeId]);
+    const film = result.rows[0];
+    res.status(200).json(film);
+    if (!film) {
+      throw new ClientError(404, `gradeId not found`);
+    }
+  } catch (err) {
+    next(err);
+  }
+});
+
+app.post('/api/grades', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+    if (!name || !course || !score) {
+      throw new ClientError(400, 'name, course, and grade are required');
+    }
+    if (score < 0 && score > 100) {
+      throw new ClientError(400, 'invalid grade');
+    }
+    if (!Number.isInteger(Number(score))) {
```

Comment:
You should check that it's an integer before you check if it's >0 or <100. That way, if it's not an integer you'll get that error first.